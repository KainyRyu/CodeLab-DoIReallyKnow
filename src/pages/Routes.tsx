import React from 'react';
import { Route, Routes as ReactRouterRoutes, Navigate } from 'react-router-dom';
import MainPage from './MainPage';

export default function Routes() {
  return (
    <ReactRouterRoutes>
      <Route path="/" element={<MainPage />} />
    </ReactRouterRoutes>
  );
}
