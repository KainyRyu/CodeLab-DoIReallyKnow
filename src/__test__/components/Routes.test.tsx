import React from 'react';
import { render, screen } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import '@testing-library/jest-dom';

import Routes from '../../pages/Routes';

describe('<Routes />', () => {
  it('renders component correctly', () => {
    const history = createMemoryHistory();
    history.push('/');

    const { container } = render(
      <Router location={history.location} navigator={history}>
        <Routes />
      </Router>
    );

    expect(container).toMatchSnapshot();
  });
  it('changes path', () => {
    const history = createMemoryHistory();
    history.push('/');

    render(
      <Router location={history.location} navigator={history}>
        <Routes />
      </Router>
    );
  });
});
