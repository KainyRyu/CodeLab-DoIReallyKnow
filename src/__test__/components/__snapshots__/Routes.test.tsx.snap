// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> renders component correctly 1`] = `
<div>
  <div>
    <h1>
      MainPage
    </h1>
  </div>
</div>
`;
